specification/dimensions/tags.md

```diff
@@ -0,0 +1,76 @@
+# Tags
+
+The Tags column represents the set of finalized user-defined and/or provider-defined tags assigned to Tag Sources.  Tags are commonly used for scenarios like adding business context to billing data to identify and accurately allocate charges.
+
+A tag becomes 'finalized' when a single value is selected from a set of possible tag values assigned to the tag key at different levels of a resource grouping hierarchy (if supported by the Provider).
+
+The Tags column adheres to the following requirements:
+
+* The Tags column MUST contain user-defined and/or provider-defined tags.
+* The Tags column MUST only contain finalized tags.
+* The Tags column MUST be in Key-Value Format.
+* A Tag without a specified value MUST have its value set to null.
+* Providers MUST NOT alter user-defined Tag keys or values.
+
+Provider-defined Tags additionally adhere to the following requirements:
+
+* Provider-defined tags MUST be prefixed with a provider-specified tag key prefix.
+* Providers SHOULD publish all provider-specified tag key prefixes within their respective documentation.
+
+## Provider-Defined vs. User-Defined Tags
+
+The following is an example of one user-defined tag and one provider-defined tag, respectively, with tag key, `foo`.  The first tag, which is user-defined, is not prefixed. The second tag is prefixed with marketplace/ which the provider has specified as a reserved tag key prefix.
+
+```json
+    {
+        "foo":"bar",
+        "marketplace/foo": "bar"
+    }
+```
+
+## Finalized Tags
+
+A tag can either be static or dynamic. If a tag is static, its value is immutable. If a tag is dynamic, its value is determined by a set of predefined user or provider rules. A finalized tag is the final result of any static or dynamic Tag.
```

Comment:
Do we need to elaborate about static or dynamic tags?

I believe the management of tags is specific for every provider.
For example in Azure, tags from subscriptions and resource groups are not automatically inherited by resources: such behavior has to be implemented by cloud users, typically by creating Azure policies.

In the end, tags are assigned by providers to charges and FOCUS only needs to specify the format for them.

Reply:
In this column, we're not exposing the universe of tags set by the provider and/or user, just the subset that tags that  are resulting, or "finalized".  If a tag, "team" has 3 different values across an Organization (`team:a`), Account (`team:b`), and Resource (`team:c`) and inheritance favors the resource, only `team:c` appears in this column.

In your comment above, we're proposing a post-1.0 column, currently called `TagDetails`, that contains every possible tag set.

Reply:
I'm still quite unhappy with the static, dynamic and final terminology. Could I make a proposal that I think keeps the details you've put in but eliminates this terminology?

```suggestion
Within a provider, tag keys may be associated with multiple values, and potentially defined at different levels within the provider, such as accounts, folders, resources and other organizational constructs. Providers MUST reduce these multiple levels of definition to a single value of Tags where one key is associated with exactly one value. The method by which this is done is up to the provider's discretion, but MUST be documented for end-users of the FOCUS-formatted data.
```

If we agree to this kind of direction, it would precipitate other changes throughout the document, but let's start here.

Reply:
I would agree to postpone the formalization of tag processing by providers and the terminology (static/dynamic/final) when the TagDetails dimension is actually introduced after FOCUS v1.0.

Meanwhile the proposition done by Joshua could be enough for v1.0.

Reply:
I really really like that suggestion, @joshk0.  Really helps lay out what the deal is.

Reply:
Ah -- I see now @cnharris10 committed a revised version of this per @udam-f2.  That looks good.